pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -199,12 +200,6 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 		allErrors = append(allErrors, err)
 	}
 	allErrors = append(allErrors, metrics.ValidateShowHiddenMetricsVersion(kc.ShowHiddenMetricsForVersion)...)
-
-	logOption := logs.NewOptions()
-	if kc.Logging.Format != "" {
-		logOption.LogFormat = kc.Logging.Format
-	}
-	allErrors = append(allErrors, logOption.Validate()...)
-
+	allErrors = append(allErrors, logs.ValidateLoggingConfiguration(&kc.Logging, field.NewPath("logging")).ToAggregate())
```

Comment:
```suggestion
	allErrors = append(allErrors, logs.ValidateLoggingConfiguration(&kc.Logging, field.NewPath("logging"))...)
```

Reply:
Not sure what I'm doing wrong here, but you suggestion doesn't work.
```
# k8s.io/kubernetes/pkg/kubelet/apis/config/validation [k8s.io/kubernetes/pkg/kubelet/apis/config/validation.test]
pkg/kubelet/apis/config/validation/validation.go:203:20: cannot use logs.ValidateLoggingConfiguration(&kc.Logging, field.NewPath("logging")) (type field.ErrorList) as type []error in append
```

I also tried 
```
allErrors = append(allErrors, logs.ValidateLoggingConfiguration(&kc.Logging, field.NewPath("logging")).ToAggregate().ToErrs()...)
```
but this panics on no errors.

Reply:
hmm… not sure about the type mistmatches… normally the allErrors we're appending to is of the same type. 

I'm mostly trying to avoid flattening all the logging validation errors into a single error that gets aggregated with individual errors for the other fields in the config.

Reply:
Sure, I think problem is related to difference between error handling between:
* Option validation using `[]error`  
* Config validation using `field.ErrorList`

Should be solved when we start migrating to using Config.

Reply:
should be able to use this to keep the individual errors intact:

`append(allErrors, logs.ValidateLoggingConfiguration(&kc.Logging, field.NewPath("logging")).ToAggregate().Errors()...)`

Reply:
This doesn't work as I mentioned above
```
--- FAIL: TestValidateKubeletConfiguration (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xc91f22]

goroutine 9 [running]:
testing.tRunner.func1.2(0xd49820, 0x14dbdb0)
	/home/siarkowicz/.gvm/gos/go1.16/src/testing/testing.go:1144 +0x332
testing.tRunner.func1(0xc0001c6780)
	/home/siarkowicz/.gvm/gos/go1.16/src/testing/testing.go:1147 +0x4b6
panic(0xd49820, 0x14dbdb0)
	/home/siarkowicz/.gvm/gos/go1.16/src/runtime/panic.go:965 +0x1b9
k8s.io/kubernetes/pkg/kubelet/apis/config/validation.ValidateKubeletConfiguration(0xc0000efb38, 0xc0002705a0, 0xe44a0a)
	/home/siarkowicz/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/apis/config/validation/validation.go:206 +0xda2
k8s.io/kubernetes/pkg/kubelet/apis/config/validation.TestValidateKubeletConfiguration(0xc0001c6780)
	/home/siarkowicz/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/apis/config/validation/validation_test.go:69 +0x307
testing.tRunner(0xc0001c6780, 0xe9eb40)
	/home/siarkowicz/.gvm/gos/go1.16/src/testing/testing.go:1194 +0xef
created by testing.(*T).Run
	/home/siarkowicz/.gvm/gos/go1.16/src/testing/testing.go:1239 +0x2b3
FAIL	k8s.io/kubernetes/pkg/kubelet/apis/config/validation	0.046s
FAIL
make: *** [Makefile:184: test] Error 1
```

NewAggregate function returns nil if there are no errors. which means it cannot be transformed further using `.Errors()`

Reply:
can you do a length check first?

```go
if errs := logs.ValidateLoggingConfiguration(&kc.Logging, field.NewPath("logging")); len(errs) > 0 {
  append(allErrors, errs.ToAggregate().Errors()...)
}
```

we want to keep all the errors individual

Reply:
This should have same results, but it might be more readable.

Reply:
Done